src/Gui/NaviCube.cpp

```diff
@@ -131,112 +96,103 @@ class NaviCubeImplementation : public ParameterGrp::ObserverType {
     int pickFace(short x, short y);
     bool inDragZone(short x, short y);
 
+    void prepare();
     void handleResize();
     void handleMenu();
 
     void setHilite(int);
 
-    void initNaviCube(QtGLWidget*);
-    void addFace(float gap, const Vector3f&, const Vector3f&, int, int, int, bool flag = false);
+    void addCubeFace(const Vector3f&, const Vector3f&, int, int);
 
-    GLuint createCubeFaceTex(QtGLWidget* gl, float gap, const char* text, int shape);
-    GLuint createButtonTex(QtGLWidget*, int);
-    GLuint createMenuTex(QtGLWidget*, bool);
+    void addButtonFaceTex(QtGLWidget*, int);
+    void addButtonFace(int);
 
     SbRotation setView(float, float) const;
     SbRotation rotateView(SbRotation, int axis, float rotAngle, SbVec3f customAxis = SbVec3f(0, 0, 0)) const;
     void rotateView(const SbRotation&);
 
     QString str(const char* str);
-    char* enum2str(int);
     QMenu* createNaviCubeMenu();
 
 public:
-    enum { //
-        TEX_FRONT = 1, // 0 is reserved for 'nothing picked'
-        TEX_REAR,
-        TEX_TOP,
-        TEX_BOTTOM,
-        TEX_LEFT,
-        TEX_RIGHT,
-        TEX_FRONT_FACE,
-        TEX_CORNER_FACE,
-        TEX_EDGE_FACE,
-        TEX_FRONT_TOP,
-        TEX_FRONT_BOTTOM,
-        TEX_FRONT_LEFT,
-        TEX_FRONT_RIGHT,
-        TEX_REAR_TOP,
-        TEX_REAR_BOTTOM,
-        TEX_REAR_LEFT,
-        TEX_REAR_RIGHT,
-        TEX_TOP_LEFT,
-        TEX_TOP_RIGHT,
-        TEX_BOTTOM_LEFT,
-        TEX_BOTTOM_RIGHT,
-        TEX_BOTTOM_RIGHT_REAR,
-        TEX_BOTTOM_FRONT_RIGHT,
-        TEX_BOTTOM_LEFT_FRONT,
-        TEX_BOTTOM_REAR_LEFT,
-        TEX_TOP_RIGHT_FRONT,
-        TEX_TOP_FRONT_LEFT,
-        TEX_TOP_LEFT_REAR,
-        TEX_TOP_REAR_RIGHT,
-        TEX_ARROW_NORTH,
-        TEX_ARROW_SOUTH,
-        TEX_ARROW_EAST,
-        TEX_ARROW_WEST,
-        TEX_ARROW_RIGHT,
-        TEX_ARROW_LEFT,
-        TEX_DOT_BACKSIDE,
-        TEX_VIEW_MENU_ICON,
-        TEX_VIEW_MENU_FACE
+    enum {
```

Comment:
You could consider making this an enum class, and dropping the `PID_` prefix -- as a general rule the C++ Core Guidelines encourage the use of enum class wherever possible (but I haven't looked to see if you need to use this as an int or things like that, so this is just an option to look into).

Reply:
Sounds good to me. It is used to color code the surfaces of the cube to find which one is hovered by the pointer but I can cast to int in that case

Reply:
Do you want to do that as part of this PR, or as a follow-up?

Reply:
@chennes It's done. Should I squash those commits to one?

Reply:
Now it passed. There was a problem triggered by the identifier OUT in the Windows build. Seems there is a macro triggered by it, I'm not sure. Not a very good name on a macro in that case? I switched to CamelCase